lib/parser/source/comment/associator.rb

```diff
@@ -188,7 +188,7 @@ def associate_and_advance_comment(node)
 
       def advance_through_directives
         # Skip shebang.
-        if @current_comment && @current_comment.text =~ /^#!/
+        if @current_comment && @current_comment.text.start_with?('#!')
```

Comment:
Note there is a small semantic difference between them: `^#!` would match ANY line that begins with `#!` not only the first. AFAIK the `@current_comment` is only a single line so the difference should not matter?

Reply:
shebang shoud be located at first line of a file. So it always should match as `/\A#!/`, I think.